poc_iot_verifier/src/follower.rs

```diff
@@ -1,24 +1,23 @@
-use crate::{env_var, Error, Result};
-
+use crate::{traits::GatewayInfoResolver, Error, Result};
+use async_trait::async_trait;
 use helium_crypto::PublicKey;
-use helium_proto::services::{
-    follower::{self, FollowerGatewayReqV1, FollowerGatewayRespV1},
-    Channel, Endpoint,
+use helium_proto::{
+    services::follower::{FollowerGatewayReqV1, FollowerGatewayRespV1},
+    GatewayStakingMode, Region,
 };
-use helium_proto::{GatewayStakingMode, Region};
-use http::Uri;
+use node_follower::FollowerService;
 use std::ops::Not;
-use std::time::Duration;
-
-const CONNECT_TIMEOUT: Duration = Duration::from_secs(5);
-const RPC_TIMEOUT: Duration = Duration::from_secs(5);
-pub const DEFAULT_URI: &str = "http://127.0.0.1:8080";
 
-type FollowerClient = follower::Client<Channel>;
-
-#[derive(Debug, Clone)]
-pub struct FollowerService {
-    client: FollowerClient,
+#[async_trait]
+impl GatewayInfoResolver for FollowerService {
+    async fn resolve_gateway_info(&mut self, address: &PublicKey) -> Result<FollowerGatewayResp> {
+        let req = FollowerGatewayReqV1 {
+            address: address.to_vec(),
+        };
+        let proto = self.client.find_gateway(req).await?.into_inner();
+        let res: FollowerGatewayResp = FollowerGatewayResp::try_from(proto).unwrap();
```

Comment:
Should the FollowerGatewayResp struct and its various TryFrom's currently defined in poc_iot_verifier::follower not be moved to the new follower service itself ?  Perhaps other oracles currently do not require the full resp payload but still they could avail of it and take what they need from it

Reply:
That's a fair point. I propose we add that to the node-follower once another service has the need to use the `FollowerGatewayResp` struct.

Reply:
the density_scaler, and by extension the iot verifier will need this

Reply:
Ah fair enough, I'll move it over to node-follower then.

Reply:
Update [1518ee8](https://github.com/helium/oracles/pull/115/commits/1518ee8df38a015f53d850492c3ce43765a3159c)